packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart

```diff
@@ -0,0 +1,375 @@
+    switch (buttonItem.type) {
+      case ContextMenuButtonType.cut:
+        return localizations.cutButtonLabel;
+      case ContextMenuButtonType.copy:
+        return localizations.copyButtonLabel;
+      case ContextMenuButtonType.paste:
+        return localizations.pasteButtonLabel;
+      case ContextMenuButtonType.selectAll:
+        return localizations.selectAllButtonLabel;
+      case ContextMenuButtonType.custom:
+        return '';
+    }
+  }
+
+  /// Returns a List of Widgets generated by turning [buttonItems] into the
+  /// the default context menu buttons for Cupertino on the current platform.
+  ///
+  /// This is useful when building a text selection toolbar with the default
+  /// button appearance for the given platform, but where the toolbar and/or the
+  /// button actions and labels may be custom.
+  ///
+  /// Does not build Material buttons. On non-Apple platforms, Cupertino buttons
+  /// will still be used, because the Cupertino library does not access the
+  /// Material library. To get the native-looking buttons on every platform, use
+  /// use [AdaptiveTextSelectionToolbar.getAdaptiveButtons] in the Material
+  /// library.
+  ///
+  /// See also:
+  ///
+  /// * [AdaptiveTextSelectionToolbar.getAdaptiveButtons], which is the Material
+  ///   equivalent of this class and builds only the Material buttons.
+  static Iterable<Widget> getAdaptiveButtons(BuildContext context, List<ContextMenuButtonItem> buttonItems) {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.iOS:
+        return buttonItems.map((ContextMenuButtonItem buttonItem) {
+          return CupertinoTextSelectionToolbarButton.text(
+            onPressed: buttonItem.onPressed,
+            text: getButtonLabel(context, buttonItem),
+          );
+        });
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+      case TargetPlatform.macOS:
+        return buttonItems.map((ContextMenuButtonItem buttonItem) {
+          return CupertinoDesktopTextSelectionToolbarButton.text(
+            context: context,
+            onPressed: buttonItem.onPressed,
+            text: getButtonLabel(context, buttonItem),
+          );
+        });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // If there aren't any buttons to build, build an empty toolbar.
+    if ((children != null && children!.isEmpty)
+      || (buttonItems != null && buttonItems!.isEmpty)) {
+      return const SizedBox(width: 0.0, height: 0.0);
+    }
+
+    final List<Widget> resultChildren = children != null
+        ? children!
+        : getAdaptiveButtons(context, buttonItems!).toList();
+
+    return _AdaptiveTextSelectionToolbarFromChildren(
+      primaryAnchor: primaryAnchor,
+      secondaryAnchor: secondaryAnchor,
+      targetPlatform: targetPlatform,
+      children: resultChildren,
+      /*
```

Comment:
Just pointing this out, in case you forgot these comments.

Reply:
Yes I did, thank you!